### ISSUE TITLE
paramiko 2.4.2 CryptographyDeprecationWarning

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#52598]
##### SUMMARY

When using ansible with paramiko I receive CryptographyDeprecationWarning 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

paramiko

##### ANSIBLE VERSION

```paste below
$ ansible --version
ansible 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/tanner/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tanner/projects/ansible.git/ansible/lib/ansible
  executable location = /Users/tanner/projects/ansible.git/ansible/bin/ansible
  python version = 2.7.10 (default, Aug 17 2018, 19:45:58) [GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.0.42)]
```

##### CONFIGURATION

```paste below
$ ansible-config dump --only-changed
```

##### OS / ENVIRONMENT

Ansible management host: macOS 10.14.3
Ansible management hosts: Ubuntu 18.04

##### STEPS TO REPRODUCE

```yaml
Problem is not playbook related. Every playbook causes the CryptographyDeprecationWarning.
```
```shell
ansible-playbook -i testing any-playbook.yml  --user="testing" -k -K
```

##### EXPECTED RESULTS

Playbook would run without CryptographyDeprecationWarning 

##### ACTUAL RESULTS

```paste below
/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/ecdsakey.py:164: CryptographyDeprecationWarning: Support for unsafe construction of public numbers from encoded data will be removed in a future version. Please use EllipticCurvePublicKey.from_encoded_point
  self.ecdsa_curve.curve_class(), pointinfo
/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/kex_ecdh_nist.py:39: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
  m.add_string(self.Q_C.public_numbers().encode_point())
/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/kex_ecdh_nist.py:96: CryptographyDeprecationWarning: Support for unsafe construction of public numbers from encoded data will be removed in a future version. Please use EllipticCurvePublicKey.from_encoded_point
  self.curve, Q_S_bytes
/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/kex_ecdh_nist.py:111: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
  hm.add_string(self.Q_C.public_numbers().encode_point())
```

**Comment by basictheprogram** [comment]
Every so often I'll get the following error. Not sure if it's related to this issue.
```
The full traceback is:
Traceback (most recent call last):
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/executor/task_executor.py", line 143, in run
    res = self._execute()
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/normal.py", line 46, in run
    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py", line 721, in _execute_module
    self._make_tmp_path()
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py", line 291, in _make_tmp_path
    result = self._low_level_execute_command(cmd, sudoable=False)
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py", line 986, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/connection/paramiko_ssh.py", line 415, in exec_command
    chan.exec_command(cmd)
  File "/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/channel.py", line 72, in _check
    return func(self, *args, **kwds)
  File "/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/channel.py", line 257, in exec_command
    self._wait_for_event()
  File "/Users/tanner/projects/ansible.git/lib/python2.7/site-packages/paramiko/channel.py", line 1208, in _wait_for_event
    raise e
EOFError

fatal: [testing.mylab.com]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}
```

**Comment by basictheprogram** [comment]
Guessing this is a paramiko problem?

  * https://github.com/paramiko/paramiko/issues/1386
  * https://github.com/paramiko/paramiko/pull/1379

**Comment by samdoran** [comment]
This does appear to be a problem with Paramiko and not Ansible.

**Comment by cjyar** [comment]
Is there any workaround for this problem? Maybe ansible didn't directly cause the warnings, but isn't ansible responsible for choosing a version of paramiko that causes this trouble?

**Comment by tomazas** [comment]
> Is there any workaround for this problem?
Add two lines to your script start to remove the annoying CryptographyDeprecationWarning:

import warnings
warnings.filterwarnings(action='ignore',module='.*paramiko.*')

**Comment by bobf** [comment]
In case this is helpful for anyone, I used this `sed` command to hack the `ansible-playbook` script:
```
sed -i "/__metaclass__ = type/a import warnings\nwarnings.filterwarnings(action='ignore',module='.*paramiko.*')" /ansible2.7/bin/ansible-playbook
```

I run Ansible in a Docker container with `virtualenv` so this may not be ideal for your setup; in my instance I don't care if I trash system application code.

**Comment by chusiang** [comment]
Thanks for @tomazas , I can skip this warning message now.

Difference:

```
[ chusiang@el-capitan ~ ] - 18:45
(ins)$ diff /usr/local/bin/ansible-playbook /usr/local/bin/ansible-playbook.ori
44,48d43
< # https://github.com/ansible/ansible/issues/52598#issuecomment-470846069
< import warnings
< warnings.filterwarnings(action='ignore',module='.*paramiko.*')
<
<
```

Version:

```
[ chusiang@el-capitan ~ ] - 18:46
(ins)$ ansible --version
ansible 2.7.7
  config file = /Users/chusiang/.ansible.cfg
  configured module search path = ['/Users/chusiang/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.7.7/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.2 (default, Feb 21 2019, 10:59:43) [Clang 8.0.0 (clang-800.0.42.1)]
```

**Comment by jobec** [comment]
Another, less drastic workaround is setting the `PYTHONWARNINGS` environment variable to `ignore::UserWarning`

_The docs for PYTHONWARNINGS state is takes a regex for the module, but it doesn't... See https://bugs.python.org/issue34624_

**Comment by sundeepk1** [comment]
is it better to run with paramiko or native ssh ?

**Comment by ssunka** [comment]
Ignoring warnings in ansible playbook helped me to remove these messages 
vi /usr/local/bin/ansible-playbook
Add
---

#Removing warnings for ansible
import warnings
warnings.filterwarnings(action='ignore',module='.*paramiko.*')

**Comment by CirrusThink** [comment]
Mine worked only when after the _metaclass statement.
Wither that or I was editing the symlink but I doubt it.

#######################################################
from __future__ import (absolute_import, division, print_function)
__metaclass__ = type

#https://github.com/ansible/ansible/issues/52598#issuecomment-470846069
import warnings
warnings.filterwarnings(action='ignore',module='.*paramiko.*')

**Comment by cowlinator** [comment]
FYI, paramiko has released 2.5.0, which no longer uses deprecated cryptography calls.
https://github.com/paramiko/paramiko/releases/tag/2.5.0
